instrumentor/controllers/instrumentationdevice/manager.go

```diff
@@ -27,7 +27,12 @@ type workloadPodTemplatePredicate struct {
 }
 
 func (w workloadPodTemplatePredicate) Create(e event.CreateEvent) bool {
-	return false
+	// when instrumentor restarts, this case will be triggered as workloads objects are being added to the cache.
+	// in this case, we need to reconcile the workload, and guarantee that the device is injected or removed
+	// based on the current state of the cluster.
+	// if the instrumented application is deleted but the device is not cleaned,
+	// the instrumented application controller will not be invoked after restart, which is why we need to handle this case here.
+	return true
```

Comment:
Maybe we should add a check here that the odigos label is present?
Since after this change, when the instrumentor is starting this will cause it to have an event for each workload in the cluster - even those which odigos does not care about.
For those the new check added in this PR below `if apierrors.IsNotFound(err)` will be true.
That means the even for workloads which were not instrumented at all we will call `removeInstrumentationDeviceFromWorkload`.

Reply:
We need to handle workloads which has no label.

Consider the following sequence:
- label is removed from a workload which has the device
- instrumented application deleted
- instrumentor went down

When the instrumentor starts, we need to remove the device from this workload. if we add the filter you suggested, nothing will remove the device from this workload

Reply:
How about checking if there is a device present in the pod spec? Since we have the workload object here I think we can check that.